src/collection-browser.ts

```diff
@@ -1211,10 +1211,19 @@ export class CollectionBrowser
       this.preloadCollectionNames(results);
       this.updateDataSource(pageNumber, results);
     }
-    if (results.length < this.pageSize) {
+
+    // When we reach the end of the data, we can't always immediately cap off the
+    // tile count, because there may be previous pages that haven't finished loading
+    // yet. So we flag the end of the data, but let the final page load set the count.
+    // This avoids race conditions dependent on the order pages arrive in.
+    if (results.length < this.pageSize || this.endOfDataReached) {
```

Comment:
Q: is there a way to track/know how many pages one has scrolled through?

Reply:
From what I understand, `this.pagesToRender` is (roughly) a measure of how many pages have been scrolled through. It's set to 1 whenever the query changes, and is incremented each time the scroll threshold for loading a new page is reached.